api/src/main/java/org/apache/iceberg/expressions/InclusiveMetricsEvaluator.java

```diff
@@ -284,14 +284,6 @@ public <T> Boolean ltEq(BoundReference<T> ref, BoundReference<T> ref2) {
         return ROWS_CANNOT_MATCH;
       }
 
-      if (ref.type().typeId() != ref2.type().typeId()) {
-        return ROWS_MIGHT_MATCH;
-      }
-
-      if (checkLowerToUpperBounds(ref, ref2, id, id2, cmp -> cmp > 0)) {
```

Comment:
when data is equal the lower to upper bound checks return that rows cannot match when they can, removing check